File: src/main/java/at/hrechny/predictionsbot/model/Season.java
package at.hrechny.predictionsbot.model;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Null;
import java.time.Year;
import java.util.UUID;
import lombok.Data;

@Data
public class Season {

  private UUID id;

  @NotNull
  private Year year;

  @Null
  private String competition;

  private boolean active = true;

}

File: src/main/kotlin/at/hrechny/predictionsbot/model/Season.kt
package at.hrechny.predictionsbot.model

import io.micronaut.core.annotation.Introspected
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Null
import java.time.Year
import java.util.UUID

@Introspected
class Season {
    var id: UUID? = null

    @field:NotNull
    var year: Year? = null

    @field:Null
    var competition: String? = null

    var active: Boolean = true

    fun isActive(): Boolean = active
}
